tls/s2n_server_key_exchange.c

```diff
@@ -135,6 +136,42 @@ int s2n_dhe_server_key_recv_parse_data(struct s2n_connection *conn, union s2n_ke
     return 0;
 }
 
+int s2n_kem_server_key_recv_read_data(struct s2n_connection *conn, struct s2n_blob *data_to_verify, union s2n_kex_raw_server_data *raw_server_data)
+{
+    struct s2n_kem_raw_server_params *kem_data = &raw_server_data->kem_data;
+    struct s2n_stuffer *in = &conn->handshake.io;
+    const struct s2n_kem *kem = conn->secure.kem_params.negotiated_kem;
+    uint16_t key_length;
+
+    /* Keep a copy to the start of the whole structure for the signature check */
+    data_to_verify->data = s2n_stuffer_raw_read(in, 0);
+    notnull_check(data_to_verify->data);
+
+    // the server sends the named KEM again and this must match what was agreed upon during server hello
+    uint8_t named_kem;
+    GUARD(s2n_stuffer_read_uint8(in, &named_kem));
+    eq_check(named_kem, kem->kem_extension_id);
```

Comment:
You compare "name" with "id", where "name" is not a string. Consider a different variable name.